staging/src/k8s.io/dynamic-resource-allocation/deviceattribute/pci.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceattribute
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	resourceapi "k8s.io/dynamic-resource-allocation/api"
+)
+
+var (
+	bdfRegexp = regexp.MustCompile(`^([0-9a-f]{4}):([0-9a-f]{2}):([0-9a-f]{2})\.([0-9a-f]{1})$`)
+
+	qualifiedNamePCIeRoot = resourceapi.QualifiedName(resourceapi.StandardDeviceAttributePCIeRoot)
+)
+
+type StandardPCIDeviceAttributesOption func(args *StandardPCIDeviceAttributeArgs)
+
+// WithPCIDeviceAddress sets the PCI address for the PCI Bus ID
+// in BDF (Bus-Device-Function) format, e.g., "0123:45:1e.7".
+//
+// It returns a DeviceAttribute with the PCIe Root Complex information("pci<domain>:<bus>")
+// as a string value or an error if the PCI Bus ID is invalid or the root complex cannot be determined.
+//
+// ref: https://wiki.xenproject.org/wiki/Bus:Device.Function_(BDF)_Notation
+func WithPCIDeviceAddress(pciBusID string) StandardPCIDeviceAttributesOption {
+	return func(args *StandardPCIDeviceAttributeArgs) {
+		args.pciAddress = pciBusID
+	}
+}
+
+// StandardPCIDeviceAttributeArgs holds the arguments for generating standard PCI device attributes.
+type StandardPCIDeviceAttributeArgs struct {
+	pciAddress string
+	sysfs      sysfs
+}
+
+// StandardPCIDeviceAttributes returns standard device attributes for a PCI device.
+func StandardPCIDeviceAttributes(opts ...StandardPCIDeviceAttributesOption) (map[resourceapi.QualifiedName]resourceapi.DeviceAttribute, error) {
+	args := &StandardPCIDeviceAttributeArgs{
+		sysfs: sysfs(defaultSysfsRoot),
+	}
+	for _, opt := range opts {
+		opt(args)
+	}
+
+	attrs := make(map[resourceapi.QualifiedName]resourceapi.DeviceAttribute)
```

Comment:
With my suggestion to have individual pkg level methods to grab the specific attribute of interest, I think this will be simplified significantly